### PR TITLE
CDAP-2489 Fixed the invalid RunId comparison and simplified the condi…

### PR DESCRIPTION
**Pull request by sagarkapare**
…tion.
Jira: https://issues.cask.co/browse/CDAP-2489

### CODE REVIEW

**Comment by sagarkapare** [comment]
@henry-cask Please take a look. Thanks!

**Comment by sagarkapare** [comment]
Build is running here - http://builds.cask.co/browse/CDAP-DUT1818-2

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java [hunk 1]
```diff
@@ -238,10 +238,8 @@ public synchronized RuntimeInfo lookup(Id.Program programId, final RunId runId)
     List<RunRecord> activeRunRecords = store.getRuns(ProgramRunStatus.RUNNING, new Predicate<RunRecord>() {
       @Override
       public boolean apply(RunRecord record) {
-        if (record.getTwillRunId() == null) {
-          return false;
-        }
-        return twillRunIds.contains(record.getTwillRunId());
+        return record.getTwillRunId() != null
+          && twillRunIds.contains(org.apache.twill.internal.RunIds.fromString(record.getTwillRunId()));
```
Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java]: Why contains? I think should be startsWith

Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java]: `twillRunIds` is of type `Set<RunId>`. We need to check if the `twillRunId` we get from the run record exists in this set.

**Comment by chtyim** [comment]
LGTM

**Comment by sagarkapare** [comment]
Thank you @chtyim for the review. Build passed. Merging.

**Pull request merged**